ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.js

```diff
@@ -743,7 +743,7 @@ export default {
         if (data.dooms)
           name = data.dooms[data.doomCount];
         data.doomCount++;
-        if (name)
+        if (name === data.me)
```

Comment:
It might make sense to rework this trigger in general? Something like:

```js
  condition: (data) => data.dooms && data.dooms.includes(data.me),
  infoText: (_data, _matches, output) => output.text({ num: data.dooms.filter((d) => d !== null).length, player: data.ShortName(data.me) }),
```

Reply:
In fact, the `UCU Doom Init` trigger could probably be simplified significantly by just making `data.dooms` a `new Set()`, allowing all the extra logic there to be dropped, and then the `filter` in the `infoText` I suggested could be removed entirely...

Reply:
Here's a more thorough cleanup, if anyone wants to test it out? This drops the `UCU Doom Cleanup` trigger (which was firing 3 times, by the way).

All of the workarounds listed in the original trigger are no longer needed, those issues are long-since-fixed, I confirmed with the CN log I have of UCU.

```js
    {
      id: 'UCU Doom Init',
      netRegex: NetRegexes.gainsEffect({ effectId: 'D2' }),
      condition: Conditions.targetIsYou(),
      run: (data, matches) => {
        const duration = parseFloat(matches.duration);
        if (duration < 9)
          order = 1;
        else if (duration < 14)
          order = 2;
        else
          order = 3;

        data.doomNumber = order;
      },
    },
    {
      id: 'UCU Nael Cleanse Callout',
      netRegex: NetRegexes.ability({ source: 'Fang Of Light', id: '26CA', capture: false }),
      netRegexDe: NetRegexes.ability({ source: 'Lichtklaue', id: '26CA', capture: false }),
      netRegexFr: NetRegexes.ability({ source: 'Croc De Lumière', id: '26CA', capture: false }),
      netRegexJa: NetRegexes.ability({ source: 'ライトファング', id: '26CA', capture: false }),
      netRegexCn: NetRegexes.ability({ source: '光牙', id: '26CA', capture: false }),
      netRegexKo: NetRegexes.ability({ source: '빛의 송곳니', id: '26CA', capture: false }),
      condition: (data) => {
        if (!data.doomNumber)
          return false;
        data.doomCount = data.doomCount || 0;
        ++data.doomCount;
        return data.doomCount === data.doomNumber;
      },
      infoText: (data, _matches, output) => {
        const ret = output.text({ num: data.doomNumber, player: data.ShortName(data.me) });
        delete data.doomNumber;
        delete data.doomCount;
        if (ret)
          return ret;
      },
      outputStrings: {
        text: {
          en: 'Cleanse #${num}: ${player}',
          de: 'Medica #${num}: ${player}',
          fr: 'Purifié #${num}: ${player}',
          ja: '解除に番目${num}: ${player}',
          cn: '解除死宣 #${num}: ${player}',
          ko: '선고 해제 ${num}: ${player}',
        },
      },
    },
```

Reply:
nit:
```js
      infoText: (data, _matches, output) => {
        return output.text({ num: data.doomNumber, player: data.ShortName(data.me) });
      },
      run: (data) => {
        delete data.doomNumber;
        delete data.doomCount;
      },
```
Not sure of the `if (ret) return ret;`? We're also modifying doomCount in the condition, but I can't think of a great way to do that via preRun/condition that's as immediately obvious as to what's going on.

Reply:
I actually was about to delete that version when you posted your nit. Here's an updated version, including the other FIXME in the trigger above those two.

```js
    {
      id: 'UCU Nael Your Doom',
      netRegex: NetRegexes.gainsEffect({ effectId: 'D2' }),
      condition: Conditions.targetIsYou(),
      durationSeconds: (_data, matches) => {
        if (parseFloat(matches.duration) <= 6)
          return 3;

        if (parseFloat(matches.duration) <= 10)
          return 6;

        return 9;
      },
      suppressSeconds: 20,
      alarmText: (_data, matches, output) => {
        if (parseFloat(matches.duration) <= 6)
          return output.doom1();
        if (parseFloat(matches.duration) <= 10)
          return output.doom2();
        return output.doom3();
      },
      tts: (_data, matches, output) => {
        if (parseFloat(matches.duration) <= 6)
          return output.justNumber({ num: '1' });

        if (parseFloat(matches.duration) <= 10)
          return output.justNumber({ num: '2' });

        return output.justNumber({ num: '3' });
      },
      outputStrings: {
        doom1: {
          en: 'Doom #1 on YOU',
          de: 'Verhängnis #1 auf DIR',
          fr: 'Glas #1 sur VOUS',
          ja: '自分に一番目死の宣告',
          cn: '死宣一号点名',
          ko: '죽음의 선고 1번',
        },
        doom2: {
          en: 'Doom #2 on YOU',
          de: 'Verhängnis #2 auf DIR',
          fr: 'Glas #2 sur VOUS',
          ja: '自分に二番目死の宣告',
          cn: '死宣二号点名',
          ko: '죽음의 선고 2번',
        },
        doom3: {
          en: 'Doom #3 on YOU',
          de: 'Verhängnis #3 auf DIR',
          fr: 'Glas #3 sur VOUS',
          ja: '自分に三番目死の宣告',
          cn: '死宣三号点名',
          ko: '죽음의 선고 3번',
        },
        justNumber: {
          en: '${num}',
          de: '${num}',
          fr: '${num}',
          ja: '${num}',
          cn: '${num}',
          ko: '${num}',
        },
      },
    },
    {
      id: 'UCU Doom Init',
      netRegex: NetRegexes.gainsEffect({ effectId: 'D2' }),
      condition: Conditions.targetIsYou(),
      run: (data, matches) => {
        const duration = parseFloat(matches.duration);
        let order;
        if (duration < 9)
          order = 1;
        else if (duration < 14)
          order = 2;
        else
          order = 3;

        data.doomNumber = order;
      },
    },
    {
      id: 'UCU Nael Cleanse Callout',
      netRegex: NetRegexes.ability({ source: 'Fang Of Light', id: '26CA', capture: false }),
      netRegexDe: NetRegexes.ability({ source: 'Lichtklaue', id: '26CA', capture: false }),
      netRegexFr: NetRegexes.ability({ source: 'Croc De Lumière', id: '26CA', capture: false }),
      netRegexJa: NetRegexes.ability({ source: 'ライトファング', id: '26CA', capture: false }),
      netRegexCn: NetRegexes.ability({ source: '光牙', id: '26CA', capture: false }),
      netRegexKo: NetRegexes.ability({ source: '빛의 송곳니', id: '26CA', capture: false }),
      condition: (data) => {
        if (!data.doomNumber)
          return false;
        data.doomCount = data.doomCount || 0;
        ++data.doomCount;
        return data.doomCount === data.doomNumber;
      },
      infoText: (data, _matches, output) => output.text({
        num: data.doomNumber.toString(),
        player: data.ShortName(data.me),
      }),
      run: (data) => {
        delete data.doomNumber;
        delete data.doomCount;
      },
      outputStrings: {
        text: {
          en: 'Cleanse #${num}: ${player}',
          de: 'Medica #${num}: ${player}',
          fr: 'Purifié #${num}: ${player}',
          ja: '解除に番目${num}: ${player}',
          cn: '解除死宣 #${num}: ${player}',
          ko: '선고 해제 ${num}: ${player}',
        },
      },
    },
```

Reply:
Heh, and I was going to reply to myself 😛 
```js
      condition: (data) => data.doomNumber, // or data.doomNumber > 0,
      preRun: (data) => data.doomCount = (data.doomCount || 0) + 1,
      infoText: (data, _matches, output) => {
        if (data.doomCount === data.doomNumber)
          return ret = output.text({ num: data.doomNumber, player: data.ShortName(data.me) });
      },
      run: (data) => {
        if (data.doomCount === 3) {
          delete data.doomNumber;
          delete data.doomCount;
        }
      },
```
Arguable still that outputs shouldn't even be `Cleanse #: <your name>`? Might be better to promote it to alert and say `Cleanse # on YOU`?

Reply:
I tested this version in raidemulator by the way, since I had a bit of time. Seems to work fine there.

There's some other issues with some of the headmarkers in this file though that someone should probably fix, `UCU Octet Nael Marker` and the two triggers following that should probably have `delaySeconds: 0.5` since they depend on the setup trigger above which triggers off the same line.

Reply:
> Arguable still that outputs shouldn't even be `Cleanse #: <your name>`? Might be better to promote it to alert and say `Cleanse # on YOU`?

I didn't want to change the callout and have it need to be retranslated unless it was necessary, unless there's some canned translation that it could be pulled from. Either way, I'm just making suggestions on Trim's PR.